crates/commonware-node/src/db/mod.rs

```diff
@@ -0,0 +1,571 @@
+            }));
+        }
+
+        if let Some(epoch_state) = self.get_epoch::<post_allegretto::EpochState>().await? {
+            return Ok(Some(PublicOutcome {
+                epoch: epoch_state.dkg_outcome.epoch,
+                participants: epoch_state.dkg_outcome.participants,
+                public: epoch_state.dkg_outcome.public,
+            }));
+        }
+
+        if let Some(epoch_state) = self.get_epoch::<pre_allegretto::EpochState>().await? {
+            return Ok(Some(PublicOutcome {
+                epoch: epoch_state.epoch,
+                participants: epoch_state.participants,
+                public: epoch_state.public,
+            }));
+        }
+
+        Ok(None)
+    }
+
+    // ── DKG Epoch Store ─────────────────────────────────────────────────────
+
+    /// Get the current epoch state for the given hardfork regime.
+    pub async fn get_epoch<S: RegimeEpochState>(&self) -> Result<Option<S>, eyre::Error> {
+        self.get(keys::current_epoch(S::REGIME)).await
+    }
+
+    /// Set the current epoch state for the given hardfork regime.
+    pub fn set_epoch<S: RegimeEpochState>(&mut self, state: S) {
+        self.insert(keys::current_epoch(S::REGIME), state)
+    }
+
+    /// Get the previous epoch state for the given hardfork regime.
+    pub async fn get_previous_epoch<S: RegimeEpochState>(&self) -> Result<Option<S>, eyre::Error> {
+        self.get(keys::previous_epoch(S::REGIME)).await
+    }
+
+    /// Set the previous epoch state for the given hardfork regime.
+    pub fn set_previous_epoch<S: RegimeEpochState>(&mut self, state: S) {
+        self.insert(keys::previous_epoch(S::REGIME), state)
+    }
+
+    /// Remove the previous epoch state for the given hardfork regime.
+    pub fn remove_previous_epoch(&mut self, regime: HardforkRegime) {
+        self.remove(keys::previous_epoch(regime))
+    }
+
+    /// Remove the current epoch state for the given hardfork regime.
+    pub fn remove_epoch(&mut self, regime: HardforkRegime) {
+        self.remove(keys::current_epoch(regime))
+    }
+
+    /// Check if a post-allegretto epoch state exists.
+    pub async fn has_post_allegretto_state(&mut self) -> bool {
+        self.contains_key(keys::current_epoch(HardforkRegime::PostAllegretto))
+            .await
+    }
+
+    /// Check if a pre-allegretto epoch state exists.
+    pub async fn has_pre_allegretto_state(&mut self) -> bool {
+        self.contains_key(keys::current_epoch(HardforkRegime::PreAllegretto))
+            .await
+    }
+
+    /// Commit all buffered writes to the store and sync.
+    ///
+    /// This serializes all pending writes, persists them to storage,
+    /// and updates the shared cache with the new values.
+    pub async fn commit(self) -> Result<(), eyre::Error> {
+        if self.writes.is_empty() {
+            return Ok(());
+        }
+
+        let mut db = self.db.0.write().await;
+
+        for (key, value_opt) in self.writes {
```

Comment:
A summary of the changes performed here would be extremely cool. Maybe at debug level?